File: Signiture-Backend/kotlin/delivery/db/src/main/java/pcy/study/db/storeuser/StoreUser.java
package pcy.study.db.storeuser;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import pcy.study.db.BaseTimeEntity;
import pcy.study.db.storeuser.enums.StoreUserRole;
import pcy.study.db.storeuser.enums.StoreUserStatus;

import java.time.LocalDateTime;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class StoreUser extends BaseTimeEntity {

    @Column(nullable = false)
    private Long storeId;

    @Column(length = 100, nullable = false)
    private String email;

    @Column(length = 100, nullable = false)
    private String password;

    @Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @Enumerated(EnumType.STRING)
    private StoreUserStatus status;

    @Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @Enumerated(EnumType.STRING)
    private StoreUserRole role;

    private LocalDateTime unregisteredAt;
    private LocalDateTime lastLoginAt;

    @Builder
    private StoreUser(Long storeId, String email, String password, StoreUserRole role) {
        this.storeId = storeId;
        this.email = email;
        this.password = password;
        this.role = role;
        this.status = StoreUserStatus.REGISTERED;
    }
}

File: Signiture-Backend/kotlin/delivery/db/src/main/kotlin/pcy/study/db/storeuser/StoreUser.kt
package pcy.study.db.storeuser

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import pcy.study.db.storeuser.enums.StoreUserRole
import pcy.study.db.storeuser.enums.StoreUserStatus
import java.time.LocalDateTime

data class StoreUser @JvmOverloads constructor(

    @field:Id
    @field:GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null,

    @field:Column(nullable = false)
    var storeId: Long,

    @field:Column(length = 100, nullable = false)
    var email: String,

    @field:Column(length = 100, nullable = false)
    var password: String,

    @field:Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @field:Enumerated(EnumType.STRING)
    var status: StoreUserStatus? = StoreUserStatus.REGISTERED,

    @field:Column(length = 50, nullable = false, columnDefinition = "VARCHAR")
    @field:Enumerated(EnumType.STRING)
    var role: StoreUserRole,

    @field:CreationTimestamp
    @field:Column(updatable = false)
    private var createdAt: LocalDateTime? = null,

    @field:UpdateTimestamp
    private var updatedAt: LocalDateTime? = null,
)
